cpp_utils/include/cpp_utils/wait/DBQueue.hpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DBQueue.hpp
+ */
+
+#ifndef DBQUEUE_HPP
+#define DBQUEUE_HPP
+
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+#include <queue>
+
+namespace eprosima {
+namespace utils {
+namespace event {
```

Comment:
I think this file should not be here. Create another directory called queue and put it there. 

I'd also refactor this class to fit the same code style as the rest of the classes of this library.